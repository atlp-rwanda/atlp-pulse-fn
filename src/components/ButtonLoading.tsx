/* eslint-disable */
import React from 'react';
import { useTranslation } from 'react-i18next';

interface Props {
  style?: any;
}
const ButtonLoading: React.FC<Props> = ({ style }) => {
  const { t } = useTranslation();
  return (
    <div
      className={`flex w-fit py-1 px-3 bg-primary justify-center font-bold ${style}`}
    >
      <div className="flex w-full gap-2 justify-center items-center">
        <div className="loader" />
        <button className=" text-white font-bold" disabled>
          {t('Processing...')}
        </button>
      </div>
    </div>
  );
};

export default ButtonLoading;
