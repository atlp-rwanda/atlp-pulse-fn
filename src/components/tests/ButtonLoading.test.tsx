import React from 'react';

import renderer from 'react-test-renderer';
import ButtonLoading from '../ButtonLoading';

describe('Button Loading', () => {
  it('Renders the button', () => {
    const elem = renderer.create(<ButtonLoading style="md" />).toJSON();
    expect(elem).toMatchSnapshot();
  });
});
