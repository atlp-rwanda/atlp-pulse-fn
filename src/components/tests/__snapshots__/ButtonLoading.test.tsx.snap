// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Loading Renders the button 1`] = `
<div
  className="flex w-fit py-1 px-3 bg-primary justify-center font-bold md"
>
  <div
    className="flex w-full gap-2 justify-center items-center"
  >
    <div
      className="loader"
    />
    <button
      className=" text-white font-bold"
      disabled={true}
    >
      Processing...
    </button>
  </div>
</div>
`;
